backend/fms_core/tests/test_services/sample_next_step_test.py

```diff
@@ -292,6 +292,61 @@ def test_move_sample_to_next_step(self):
         self.assertEqual(StepHistory.objects.filter(study=study_B, process_measurement__source_sample=sample_in, step_order__step=step).count(), 1)
         self.assertEqual(StepHistory.objects.filter(study=study_C, process_measurement__source_sample=sample_in, step_order__step=step).count(), 1)
 
+    def test_move_sample_to_next_step_repeat_step(self):
+        container1 = Container.objects.create(**create_sample_container(kind='tube', name='TestTube01_1', barcode='T123456_1'))
+        sample_in = create_fullsample(name="TestSampleNextStep_in",
+                                      alias="TestSampleNextStep_in",
+                                      volume=5000,
+                                      individual=self.valid_individual,
+                                      sample_kind=self.sample_kind_DNA,
+                                      container=container1,
+                                      project=self.project)
+        
+        step = Step.objects.get(name="Normalization (Sample)")
+        for derived_by_sample in sample_in.derived_by_samples.all():
+            derived_by_sample.project_id = self.project.id
+            derived_by_sample.save()
+                
+        letter_B = "B"
+        start = 3
+        end = 7
+        study_B = Study.objects.create(letter=letter_B,
+                                       project=self.project,
+                                       workflow=self.workflow_pcr_free,
+                                       start=start,
+                                       end=end)
+
+        letter_C = "C"
```

Comment:
Do you actually need a `letter_C` variable, especially when you already have a `study_C` variable?

Reply:
It is so that we can have on hand a not hard-coded way to reference the study in tests (which has a letter as key). It is not particularly useful but it seem a good idea nevertheless.